### ISSUE TITLE
data-main and being able to guarantee config loaded before beginning to require modules

### ISSUE THREAD

**Issue by ericnewton76** [jrburke/requirejs#1413]
Your documentation mentions data-main loading as being async and cannot be counted on being loaded before any inline require are encountered.

Also, I get it that requirejs uses a convention based approach so that theoretically configuration isnt neccesary.

What if you develop a method that will not execute until configuration has loaded (ie require.config has run). I get it, "if config failed to load then nothing runs!" yes... however if you're using a configuration that follow the standard convention then basically nothing will run anyways.

In summary, I suggest one of the following:
A) create a new method that guarantees configuration is loaded before executing
B) Create an alternate form of your library, same public api, but the configuration-is-loaded guarantee is built in, and the developer chooses which library to load: require.js or require-config.js. So all calls to require queue up until require.config has run and presumably the correct baseurl and various module dependencies defined have been incorporated into the requirejs runtime.
